app/src/main/java/com/example/sopt_1/presentation/home/RepositoryListAdapter.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.sopt_1.presentation.home
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.sopt_1.data.RepositoryInfo
+import com.example.sopt_1.databinding.ItemRepositoryBinding
+
+class RepositoryListAdapter : RecyclerView.Adapter<RepositoryListAdapter.RepositoryViewHolder>() {
+
+    /* 2. Adapter 는 ViewHolder 로 변경할 Data 를 가지고 있어야함 */
+    val userList = mutableListOf<RepositoryInfo>()
+
+    // 3. Adapter 는 아이템마다 ViewHolder를 만드는 방법을 정의
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RepositoryViewHolder {
+        val binding = ItemRepositoryBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+        return RepositoryViewHolder(
+            binding
+        )
+    }
+
+    // 4. Adapter 는 전체 아이템의 수를 알아야 함
+    override fun getItemCount(): Int = userList.size
+
+    // 5. Adapter 는 ViewHolder 에 Data 를 전달하는 방법을 정의
+    override fun onBindViewHolder(holder: RepositoryViewHolder, position: Int) {
+        holder.onBind(userList[position])
+    }
+
+    class RepositoryViewHolder(
+        private val binding: ItemRepositoryBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun onBind(repositoryInfo: RepositoryInfo) {
+            binding.tvRepositoryName.text = repositoryInfo.repoName
+            binding.tvRepositoryInfo.text = repositoryInfo.repoInfo
+            binding.tvLanguage.text = repositoryInfo.repoLanguage
```

Comment:
이부분은 나중에 item_repository 에서 <data></data> 안에 RepositoryInfo형 변수를 선언한다면,
예를들어 변수명을 repoData로 선언했다고치면
```suggestion
            binding.repoData = repositoryInfo
```
이렇게 간단하게도 처리가 가능할거야! 
저렇게 처리가 되면 
tvRepositoryName에서  `android:text = "@{repoData.repoName}" ` 이런식으로 사용이 가능해!

위의 방법이 어떤건지는 한번 찾아보는 것도 좋을 것 같아!!  (ps. 리사이클러뷰 데이터바인딩으로 검색해도 많이 나올거야)

Reply:
우왕 데이터바인딩도 같이 해봐야겠다 ㅎㅎ 고마웡🤍